db/db_impl/db_impl.cc

```diff
@@ -1395,6 +1395,9 @@ Status DBImpl::SyncWAL() {
       auto& log = *it;
       assert(!log.getting_synced);
       log.getting_synced = true;
+      // Size is expected to be monotonically increasing.
```

Comment:
I wonder if we can put the following in one helper function
```cpp
assert(!log.getting_synced);
log.getting_synced = true;
// Size is expected to be monotonically increasing.
assert(log.writer->file()->GetFileSize() >= log.pre_sync_size);
log.pre_sync_size = log.writer->file()->GetFileSize();
```
Looks like the above is used in `SyncWAL`, `SyncClosedLogs`,  and `PreprocessWrite` (except the logs_to_sync).

Reply:
There's still a problem I haven't figured out. `log.writer->file()->GetFileSize()` can actually still be too far ahead due to buffering in `WritableFileWriter` when `manual_wal_flush == true`. The `GetFileSize()` includes data in the `WritableFileWriter` buffer (counterintuitively since that data is not in the file).

Reply:
> There's still a problem I haven't figured out

Added `WritableFileWriter::GetFlushedSize()` and used that. I didn't want to call `FSWritableFile::GetFileSize()` since it may involve RPC, although it typically wouldn't as the file size is easily trackable in `Append()` / `PositionedAppend()`.

Reply:
> I wonder if we can put the following in one helper function

Done.